text/0107-construct-library-module-lifecycle.md

```diff
@@ -0,0 +1,146 @@
+
+* **Stage 0 activities:**
+    * **Publish package**  - a package for the module is autogenerated and published in the aws-cdk GitHub repo
+    * **Publish README** - the module has a rudimentary README doc that: 
+        * enables the [Stability:Stable] tag
+        * links to the relevant CloudFormation docs
+        * includes the basic import statement 
+    * **Update API reference** - list the module in the AWS Construct Library API reference
+    * **Create GitHub tracking issue** - create a GitHub tracking issue for the module. The primary role of the tracking issue is to: 
+        * provide an artifact that represents an entire AWS service that customers can +1 to help us prioritize, and 
+        * provide a single artifact to place on the public roadmap to indicate work in progress.
+    * **Observe** - watch KPIs (utilization, +1s, and feature requests)
+* **Exit Criteria** - AWS-CDK Construct Library modules move to stage 1: 
+    * when selected by the CDK team using the [prioritization framework](#Construct-prioritization-framework)
+    * OR, when the first community PR for a module is merged
+
+## Stage 1 - Experimental
+
+The goal of the experimental stage is to iterate on the design and functionality of a construct library module with the freedom, and understanding of customers, to make breaking changes when desirable or necessary. During this stage the  primary use cases and the set of L2 constructs required to enable them are incrementally identified, implemented, and validated. 
+
+While experimenting on the module API design and implementation, we will be as transparent as possible with customers about the work in progress. To achieve this we will use the module’s GitHub tracking issue to document use cases and construct design as they are developed in each iteration. Not all communication is required to occur in the tracking issue itself. For large and/or complex changes, we will author a Request for Comments (RFC) that outlines our design and publish it on the roadmap for feedback. We will also use pull requests to conduct construct reviews. In all cases, relevant RFCs, PRs, bugs, and feature requests that are in scope are linked in the module’s tracking issue for visibility. See [Appendix A](#Appendix-A:-GitHub-Artifacts) for an illustration of how these GitHub artifacts are used and related to each other.
+
+* **Stage 1 activities:**
+
+    * **Identify key stakeholders** - Opportunistically recruit stakeholders outside the CDK team to participate in API reviews and usability studies. Potential stakeholders include: service's product manager, service's developer advocates (if any), service's software development engineer(s) (designated by service's leadership), solutions architects who specialize in service (sourced via internal mailing lists), and AWS Developer Tools developer advocates. 
+    * **Update GitHub tracking issue** - Summarize the primary use cases and required constructs as they are identified
```

Comment:
Can we get an example of what this tracking issue would look like?

I'm also wondering if, after every stage, we should copy over the list of activities of the next stage into the Github tracking issue as a checklist, that we can check off as we complete. What do you think?

Reply:
yep, I've added creations of a tracking issue example to my list.

updating the tracking issue with the list of activities at each stage is exactly what I have in mind.